x-pack/legacy/plugins/siem/public/components/page/add_filter_to_global_search_bar/index.tsx

```diff
@@ -26,21 +25,52 @@ export const AddFilterToGlobalSearchBar = React.memo<OwnProps>(
   ({ children, filter, onFilterAdded }) => {
     const { filterManager } = useKibana().services.data.query;
 
-    const addToKql = useCallback(() => {
+    const filterForValue = useCallback(() => {
       filterManager.addFilters(filter);
+
       if (onFilterAdded != null) {
         onFilterAdded();
       }
-    }, [filter, filterManager, onFilterAdded]);
+    }, [filterManager, filter, onFilterAdded]);
+
+    const filterOutValue = useCallback(() => {
```

Comment:
I think we can simplify `filterOutValue` and 'filterValue`
```
 const filterValue = useCallback(
      (negate: boolean = false) => {
        filterManager.addFilters({
          ...filter,
          ...(negate
            ? {
                meta: {
                  ...filter.meta,
                  negate,
                },
              }
            : {}),
        });

        if (onFilterAdded != null) {
          onFilterAdded();
        }
      },
      [filterManager, filter, onFilterAdded]
    );
``` 
And use it like that for negation if needed it like that `filterValue.bind(null, true)`